custom_components/micro_air_easytouch/climate.py

```
@@ -285,48 +315,71 @@ async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+                    # Optimistic update
+                    self._state["heat_fan_mode_num"] = fan_value
+                elif self.hvac_mode == HVACMode.AUTO:
+                    changes["autoFan"] = fan_value
+                    # Optimistic update
+                    self._state["auto_fan_mode_num"] = fan_value
+                
+                self.async_write_ha_state()
+                
+                message = {"Type": "Change", "Changes": changes}
+                if not await self._data.send_command(self.hass, ble_device, message):
+                    raise HomeAssistantError(f"Failed to set fan mode to {fan_mode}")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)